src/js/sources-util.js

```diff
@@ -0,0 +1,17 @@
+import { derived, Readable } from 'svelte/store';
+import { Message } from './types.js';
+
+/**
+ * Removes duplicate messages that come from mchad sync with ytc
+ *
+ * @param {Readable<Message>} source
+ * @return {Readable<Message>}
+ */
+export function removeDuplicateMessages(source) {
+  let lastText = '';
+  return derived(source, $msg => {
+    if ($msg == null || $msg.text.trim() == lastText.trim()) return;
```

Comment:
this only checks lastText. you have to have a set and check if the text is in the set.

+ you should also check the author name, since you wouldn't want to block duplicates from independent translators.

Reply:
I only had it check `lastText` since Taishi does `Coco: *explains meme` a lot with other translations in between. Also, checking the name won't work since Taishi is known as `Taishi Ch.` on ytc and `Kamashiro Taishi` on mchad. If two translators do exactly the same text right next to each other, we should also probably not display that.